### PR TITLE
Edit 1

### PR DESCRIPTION
**Pull request by johnmckellar**

### CODE REVIEW

**Comment by johnmckellar** [comment]
Please review: Upload of rock-unit-rank.ttl and addition of Lens & NoRank, with Tongue thrown in for good measure (Micaela, please note). For NoRank, do we eliminate skos:related geof: or, under geof: create a collection of different rock units: Stratigraphic, Intrusive, Complex, Tectonometamorphic & Tectonostratigraphic RUs?

**Comment by nicholascar** [comment]
The vocab is valid RDF & VocPrez SKOS according to the VocPub validator: http://rdftools.surroundaustralia.com/validate

**Review by nicholascar — APPROVED** [review]
Approved on a technical level - not a subject matter approval!

**Review by BlakePaul — COMMENTED** [review]
Most of the definitions look OK to me.  However, under Suite, Line 725
skos:definition "A group of two or more related intrusive igneous rock units. Two or more related suites may form a supersuite, and a suite may be subdived into two or more suites (After: Gillespie & Leslie, 2021; cf. North American Stratigraphic Code, 2005; cf. Kumpulainen, 2017)."@en ;

should it be “and a suite may be subdivided into two or more subsuites”

**Comment by johnmckellar** [comment]
@BlakePaul 
Good spotting Paul: Yes, subsuites. Thanks for the review!!
Cheers,
John

File: rock-unit-rank.ttl
Comment (KellyVance) [inline rock-unit-rank.ttl:550]:
```suggestion
  skos:related geof:ComplexRockUnit ;
```

Assume this is geof: as per the other references

**Review by KellyVance — CHANGES_REQUESTED** [review]
Minor edits to references.
I'm working on the assumption that all the geof: references are present in the accompanying vocabulary.

I will try to upload this to the UAT environment of VocPrez.
But for whatever reason our instances seem to struggle rendering https://linked.data voabularies (as opposed to http://)

File: rock-unit-rank.ttl
Comment (johnmckellar) [reply rock-unit-rank.ttl:550]: It’s my proposal to add ComplexRockUnit to geof, along with others, e.g, IntrusiveRockUnit, TectonometamoprphicRockUnit, etc. 

For ComplexRockUnit, we would then have:
geof:ComplexRockUnit a skos:Concept ; 
skos:altLabel "complex unit"@en,
    "composite unit"@en ;
etc.

Other changes to geof include, e.g., 'geof:tectonostratigraphicEntity a skos:Concept ;' being changed to 'geof:TectonostratigraphicRockUnit a skos:Concept ;'

I originally had ComplexRockUnit, IntrusiveRockUnit, etc. in this current rur vocab, but, in the way we have constructed geof, it belongs there, with the other types of rock units (lithostratigraphic, tectonostratigraphic, etc.), unless we undergo a major reconstruct.

Comment (KellyVance) [reply rock-unit-rank.ttl:550]: That's fine. You just need to make sure the rur: and geof: prefixes are pointing to the right place, e.g. change rur:ComplexRockUnit to geof:ComplexRockUnit

**Comment by predam1** [comment]
John,

Attached are my comments. Sorry, I can’t remember how do it through github.

I would suggest to add the BRUCS rank in all definitions. This system is extensively mentioned in ‘reference’ or ‘provenance’ but only once (see IntrusionCentre) in ‘definition’. I found it useful in the context of rank.

I also have a general comment regarding joining tables. I assume that the ultimate aim is to join rock_units with rock_rank and make sure each unit has a rank or a ‘no rank’ attribute. In this case, some units may require 2 or 3 ranks, one unit may be an intrusion but also part of a complex, or a dyke but also part of a caldera. So the relationship unit-rank may become one-to-many, rather than one-to-one as it was in the past. This is one of the reasons I suggest adding the BRUCS rank in definitions; it would assist future analyses and updates. The future state may include several rank columns, rank 1, rank 2, etc. to fully characterise a rock unit.

Regards,
Micaela

**Comment by KellyVance** [comment]
This is a very good point by Micaela. The relationships in many cases will be 1 to many so the Features table will be able to link to any number of attributes in the Properties table.

@BlakePaul and @predam1 may also have some input in that some features are scale dependent, which is part of what I was struggling with in reconciling mapping units. But really mapping units are a Composite feature.
e.g. A composite Mapping Unit may have a 'no rank' designation, but it will have feature relationships (hasPart) to the subsidiary units that it is comprised of that will have their own ranks. That doesn't alter the review of the rock-unit-rank vocabulary but should be considered and discussed as part of our migration of units into GeoProperties once these vocabularies are established. And may drive the inclusion of a Composite Unit in the geofeatures vocabulary (if not already there), pending advice from yourselves.





> John, Attached are my comments. Sorry, I can’t remember how do it through github. I would suggest to add the BRUCS rank in all definitions. This system is extensively mentioned in ‘reference’ or ‘provenance’ but only once (see IntrusionCentre) in ‘definition’. I found it useful in the context of rank. I also have a general comment regarding joining tables. I assume that the ultimate aim is to join rock_units with rock_rank and make sure each unit has a rank or a ‘no rank’ attribute. In this case, some units may require 2 or 3 ranks, one unit may be an intrusion but also part of a complex, or a dyke but also part of a caldera. So the relationship unit-rank may become one-to-many, rather than one-to-one as it was in the past. This is one of the reasons I suggest adding the BRUCS rank in definitions; it would assist future analyses and updates. The future state may include several rank columns, rank 1, rank 2, etc. to fully characterise a rock unit. Regards, Micaela From: johnmckellar ***@***.***> Sent: Thursday, 3 March 2022 4:08 PM To: geological-survey-of-queensland/vocabularies Cc: Micaela Preda; Review requested Subject: Re: [geological-survey-of-queensland/vocabularies] Edit 1 (PR #423) @BlakePaul<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_BlakePaul&d=DwMCaQ&c=tpTxelpKGw9ZbZ5Dlo0lybSxHDHIiYjksG4icXfalgk&r=mhGoeoDgRI97hplQp-uSLPQD3ojy-jwdRDVfAh4SEJw&m=f1cORPD9l031VHluUb2xyVTvLgZRhzaHF7fyh4rkvWM&s=zzjMf1jH1WgwhuyQcHb7Uqo0gbJOwsv0IqwoUBRNyl0&e=> Good spotting Paul: Yes, subsuites. Thanks for the review!! Cheers, John — Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_geological-2Dsurvey-2Dof-2Dqueensland_vocabularies_pull_423-23issuecomment-2D1057709322&d=DwMCaQ&c=tpTxelpKGw9ZbZ5Dlo0lybSxHDHIiYjksG4icXfalgk&r=mhGoeoDgRI97hplQp-uSLPQD3ojy-jwdRDVfAh4SEJw&m=f1cORPD9l031VHluUb2xyVTvLgZRhzaHF7fyh4rkvWM&s=a2skN-3NhkYUYuG3x3tHfAP78XLuslUmonW8IiD8RLY&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ALW33U2KI2AHAU6JWX5IIDLU6BJLHANCNFSM5PGHLC7A&d=DwMCaQ&c=tpTxelpKGw9ZbZ5Dlo0lybSxHDHIiYjksG4icXfalgk&r=mhGoeoDgRI97hplQp-uSLPQD3ojy-jwdRDVfAh4SEJw&m=f1cORPD9l031VHluUb2xyVTvLgZRhzaHF7fyh4rkvWM&s=pKo5n0UoXUI37wk8m054Fwj3CG_BSk6SxF64axQ0JCM&e=>. Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=tpTxelpKGw9ZbZ5Dlo0lybSxHDHIiYjksG4icXfalgk&r=mhGoeoDgRI97hplQp-uSLPQD3ojy-jwdRDVfAh4SEJw&m=f1cORPD9l031VHluUb2xyVTvLgZRhzaHF7fyh4rkvWM&s=MHlX36k1a4gWNzEW5PIqmnFVCRsBWOoNMCEZNRmYVu0&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=tpTxelpKGw9ZbZ5Dlo0lybSxHDHIiYjksG4icXfalgk&r=mhGoeoDgRI97hplQp-uSLPQD3ojy-jwdRDVfAh4SEJw&m=f1cORPD9l031VHluUb2xyVTvLgZRhzaHF7fyh4rkvWM&s=WKt6nUSM3Bw6K2RJu-B8-GKaE0099lfWa5xWm0JxAZU&e=>. You are receiving this because your review was requested.Message ID: ***@***.***>
> […](#)
> ------------------------------ The information in this email together with any attachments is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. There is no waiver of any confidentiality/privilege by your inadvertent receipt of this material. Any form of review, disclosure, modification, distribution and/or publication of this email message is prohibited, unless as a necessary part of Departmental business. If you have received this message in error, you are asked to inform the sender as quickly as possible and delete this message and any copies of this message from your computer and/or your computer system network. ------------------------------

**Comment by johnmckellar** [comment]
Hi Micaela,

Thanks for the feedback.

In one of my more recent approaches to the construction of this vocab, I did include levels of rank, but I did this with broader-narrower relationships. Nick advised that this was an incorrect approach and I removed it (largely, except for where it was appropriate), flattening out the structure of the vocab, as he advised.

With the vocab in its present form, an intrusion  (per Gillespie & Leslie: a unit term at Rank 6 for any intrusion whose
form is not known or does not conform to one of those denoted by another unit term) can be that alone and can also be part of a caldera complex, a centre or cluster, etc. So, does not the vocab in its present form cover this?

If you look at the complexity of Gillespie & Leslie's Figure 2 (the key features of BRUCS) and just looking at intrusions alone, 'centre' and 'cluster' both sit at Rank 4;  lopolith, pluton, ring-intrusion, and swarm, at Rank 5; and cone-sheet, diatreme, dyke, intrusion, laccolith, neck, pipe, plug, ring-dyke, sheet, sill, vein and vent, at Rank 6.

These rock units may be associated in many, many different combinations and not all ranks, even those at higher levels (subsuite, suite, supersuite) may be present in any combination of related rock co-occurrences, as advised by G&L. The complexity of this is shown by their figures 3 to 10. Initially, for intrusions at Rank 4, I had combined centre-cluster and left unranked everything below this, to overcome this complexity.

But, G&L's scheme almost defies modelling. Far more complex mereological coding may need to be introduced. I am not sure there is a simple answer to this, to account for all possible/potential combinations. Nick will need to advise on this one.

Cheers,

John



________________________________

File: rock-unit-rank.ttl
Comment (johnmckellar) [reply rock-unit-rank.ttl]: No, the geof changes have not yet been made, but are ready to go. I held them back pending approval of my structuring of rur.

**Comment by predam1** [comment]
John,

The vocab is well-documented and your detailed work is impressive. I’m sorry I didn’t mention that in the first place.

My comments come from my experience on statistical work, which in turn, made be very practical. I’ve spent the last 14 months being concerned with cleansing data, joining tables, assessing relationships and, given that, I reflected on the relationship between rock units and rank.

The vocab is brilliant, my comments refer to the next steps, specifically, how we are going to implement the vocab and link rank with rock units.

Regards,
Micaela

**Comment by johnmckellar** [comment]
Thank you, Micaela. I agree with you (about the next step, that is 😉), and it has always been at the back of mind. But, I would like to better deal with rank in the vocab, there being multiple issues to overcome here, and this is where Nick can assist.

It would be great to look at say a specific geological province, such as the Hunter-Bowen Orogen, and pull up rock ranks and types (as well as other geologic features) associated with it, and vice versa. 👍
________________________________

**Comment by predam1** [comment]
Totally agreed. This needs a staged approach, too complex to address everything in one go.

As far I could see, the vocab is in very good shape, although additions related to specific provinces may still be needed.
However, we could start implementing the vocab as is and start allocating rank 6 first – see how that goes.
Overall, I was thinking turning Gillespie & Leslie's Figures 1 and 3 to 10 into something like this:

Qld rock unit
Rank 6
Rank 5
Rank 4
Rank 3
Rank 2
Rank 1
Unit 1
sill
sill cluster




Unit 2
intrusion


complex


Unit 3
bed

formation

group


I have some ideas on how to automate the rank allocation, then we’ll need a few brainstorming sessions with experts to review the automated component and fill in the rest.

In the meantime, the vocab will evolve and with it, rank allocations to old or new rock units.

Micaela

**Review by KellyVance — APPROVED** [review]
Technically sound.
Content naively looks good, but will defer to @BlakePaul and @predam1 

Very detailed, very thorough.

**Comment by johnmckellar** [comment]
Merged, thanks everyone. Now ready for round two, whatever this may entail??

**Pull request merged**
